api/v1/configurationpolicy_types.go

```diff
@@ -211,120 +175,133 @@ const (
 	RecordDiffNone RecordDiff = "None"
 )
 
-// ConfigurationPolicyStatus defines the observed state of ConfigurationPolicy
-type ConfigurationPolicyStatus struct {
-	ComplianceState   ComplianceState  `json:"compliant,omitempty"`         // Compliant/NonCompliant/UnknownCompliancy
-	CompliancyDetails []TemplateStatus `json:"compliancyDetails,omitempty"` // reason for non-compliancy
-	// An ISO-8601 timestamp of the last time the policy was evaluated
-	LastEvaluated string `json:"lastEvaluated,omitempty"`
-	// The generation of the ConfigurationPolicy object when it was last evaluated
-	LastEvaluatedGeneration int64 `json:"lastEvaluatedGeneration,omitempty"`
-	// List of resources processed by the policy
-	RelatedObjects []RelatedObject `json:"relatedObjects,omitempty"`
-}
+// ObjectTemplate describes how an object should look
+type ObjectTemplate struct {
+	ComplianceType         ComplianceType         `json:"complianceType"`
+	MetadataComplianceType MetadataComplianceType `json:"metadataComplianceType,omitempty"`
 
-// CompliancePerClusterStatus contains aggregate status of other policies in cluster
-type CompliancePerClusterStatus struct {
-	AggregatePolicyStatus map[string]*ConfigurationPolicyStatus `json:"aggregatePoliciesStatus,omitempty"`
-	ComplianceState       ComplianceState                       `json:"compliant,omitempty"`
-	ClusterName           string                                `json:"clustername,omitempty"`
+	// ObjectDefinition defines required fields, in whole or in part, to be compared with objects on
+	// the cluster.
+	//
+	// +kubebuilder:pruning:PreserveUnknownFields
+	ObjectDefinition runtime.RawExtension `json:"objectDefinition"`
+
+	// RecordDiff specifies whether (and where) to log the diff between the object on the
+	// cluster and the objectDefinition in the policy. Defaults to "None".
```

Comment:
Maybe it is better to write the "log" option to this explanation?!!

Reply:
Good point--updated!